### PR TITLE
Bug 1911443: Validate SSH public key in cloud init form

### PR DESCRIPTION
**Pull request by yaacov**
Validate SSH public key in cloud init form:
  - Show an error if auth key is invalid
  - Show a hint about SSH key format
  
Screenshots:
Hint:
screenshot-localhost_9000-2020 12 30-12_15_25
Invalid key:
screenshot-localhost_9000-2020 12 30-12_15_52
Valid key:
screenshot-localhost_9000-2020 12 30-12_16_39

### CODE REVIEW

**Comment by yaacov** [comment]
@glekner @metalice please review

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/advanced-tab-validation.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  const isForm = iGetCloudInitValue(state, id, CloudInitField.IS_FORM);
+  const authKeys = iGetCloudInitValue(state, id, CloudInitField.AUTH_KEYS);
+
+  // t('kubevirt-plugin~Invalid SSH public key format in Cloud Init settings.')
+  const errorKey =
+    getAuthKeyError(isForm, authKeys) &&
+    'kubevirt-plugin~Invalid SSH public key format in Cloud Init settings.';
+  const hasAllRequiredFilled = !errorKey;
+  const isValid = !errorKey;
```
Comment (metalice) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/advanced-tab-validation.ts:25]: same?

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/advanced-tab-validation.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  if (
+    checkTabValidityChanged(
+      state,
+      id,
+      VMWizardTab.ADVANCED_CLOUD_INIT,
+      isValid,
+      hasAllRequiredFilled,
+      errorKey,
```
Comment (metalice) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/advanced-tab-validation.ts:34]: same as above

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/advanced-tab-validation.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      null,
+    )
+  ) {
+    dispatch(
+      vmWizardInternalActions[InternalActionType.SetTabValidity](
+        id,
+        VMWizardTab.ADVANCED_CLOUD_INIT,
+        isValid,
+        hasAllRequiredFilled,
+        errorKey,
```
Comment (metalice) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/advanced-tab-validation.ts:44]: same as above

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/cloud-init-tab/cloud-init-tab.tsx [hunk 4]
```diff
@@ -154,6 +166,13 @@ const CloudInitFormRows: React.FC<CloudInitFormRowsProps> = ({
             </Split>
           );
         })}
+        <div className="pf-c-form__helper-text" aria-live="polite">
+          {t('kubevirt-plugin~Authorized keys must folow the SSH Pulic key format,')}{' '}
```
Comment (metalice) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/cloud-init-tab/cloud-init-tab.tsx:170]: consider use css instead of {" "} ?

**Pull request merged**